pkg/autoscaler/stats_scraper.go

```diff
@@ -210,14 +219,25 @@ func (s *ServiceScraper) Scrape() (Stat, error) {
 	}, nil
 }
 
-// tryScrape runs a single scrape and checks if this pod wasn't already scraped
-// against the given already scraped pods.
-func (s *ServiceScraper) tryScrape(scrapedPods *sync.Map) (Stat, error) {
+// tryScrape runs a single scrape and checks if this pod is young enough or
+// this pod wasn't already scraped against the given already scraped pods.
+func (s *ServiceScraper) tryScrape(scrapedPods, youngPods *sync.Map, sampleSize int) (Stat, error) {
 	stat, err := s.sClient.Scrape(s.url)
 	if err != nil {
 		return emptyStat, err
 	}
 
+	// Check the pod life span, if we have enough pod diversity.
+	// TODO(vagababov): remove `ProcessUptime > 0` check in 0.14.
+	if sampleSize >= minSampleSizeToConsiderAge && stat.ProcessUptime > 0 && stat.ProcessUptime < youngPodCutOffSecs {
+		// If the pod has already been returned once, then permit it. Since
```

Comment:
I'm not sure about this fallback logic. I wonder if we should do something akin to:

``` golang
scraped := &sync.Map{}
oldEnoughStats := make(chan Stat, scrapeSize)
tooYoungStats := make(chan Stat, readyPodCount) // maybe make this smaller? min(scrapeSize*2, readyPodCount) for example

for ... {
  stat, err := tryScrape(scraped)
  // err stuff
  if stat.ProcessUptime > youngPodCutOffSecs {
    select {
    case oldEnoughStats <- stat:
    default:
      return  // yay, we got enough
    }
  } else {
    select {
    case tooYoungStats <- stat:
    default:
      // ignored
    }
  }
}

// range over oldEnough stats and add to the averages
// if successCount < scrapeSize, fill up with stats from the tooYoungStats channel, potentially sorted by age to bias for the oldest of that set as well

// if both are too small, error with "too small sample size" error as today
```

Pardon this horrible mess of golang and pseudocode, hope it gets the point across 😂. I feel like that'd try harder to ignore the younger pods and only fall back to them if necessary.

Reply:
This is much harder in real life.
But I made it work :)